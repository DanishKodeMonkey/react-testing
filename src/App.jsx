import React, { useState } from 'react';

import './App.css';

const App = () => {
    const [heading, setHeading] = useState('danish Kode Monkey');

    const clickHandler = () => {
        setHeading('Ravenous Ravens');
    };

    return (
        <>
            <button
                type='button'
                onClick={clickHandler}
            >
                Click me
            </button>
            <h1>{heading}</h1>
        </>
    );
};

export default App;
