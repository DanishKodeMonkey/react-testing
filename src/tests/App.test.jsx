/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import App from '../App';

describe('App component', () => {
    it('renders danish Kode Monkey', () => {
        // destructure render to obtain a container for the test
        const { container } = render(<App />);

        // test against snapshot of expected render, a HTML representation of the App component
        expect(container).toMatchSnapshot();
    });

    // note the asynchronous function
    it('renders Ravenous Ravens after button click', async () => {
        // Create a instance of a user event, this will simulate a user interaction
        const user = userEvent.setup();

        // render an instance of the component
        render(<App />);

        // fetch the button element of the component
        const button = screen.getByRole('button', { name: 'Click me' });

        // await a effect to finish from user.click method on button
        await user.click(button);

        // expected result from click, heading textContent should match Ravenous Ravens regex
        expect(screen.getByRole('heading').textContent).toMatch(
            /Ravenous Ravens/i
        );
    });
});
