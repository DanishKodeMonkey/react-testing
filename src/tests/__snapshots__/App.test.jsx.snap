// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders danish Kode Monkey 1`] = `
<div>
  <button
    type="button"
  >
    Click me
  </button>
  <h1>
    danish Kode Monkey
  </h1>
</div>
`;
